confluent_kafka/src/Consumer.c

```diff
@@ -391,12 +391,18 @@ static PyObject *Consumer_poll (Handle *self, PyObject *args,
 
 static PyObject *Consumer_close (Handle *self, PyObject *ignore) {
 	CallState cs;
+	int raise = 0;
 
 	CallState_begin(self, &cs);
 
 	rd_kafka_consumer_close(self->rk);
 
-	if (!CallState_end(self, &cs))
+	raise = !CallState_end(self, &cs);
```

Comment:
My understanding of the behavior here is that we essentially want to do the equivalent of `Consumer_dealloc`, but as far as I can tell, the `rd_kafka_destroy` is only a fraction of what should actually be done? Why isn't the full `Consumer_clear` (or more completely, `Consumer_dealloc`) performed? In particular, since the full `Consumer_clear` is not performed, doesn't this mean we'll leave some callbacks un-decref'd?